TestFoundation/HTTPServer.swift

```diff
@@ -207,7 +207,7 @@ class _HTTPServer {
             deadlineTime = .now()
         }
 
-        DispatchQueue.main.asyncAfter(deadline: deadlineTime) {
+        DispatchQueue.global().asyncAfter(deadline: deadlineTime) {
```

Comment:
Can we split up the data fix itself from the change to the URLSession test code?

Reply:
No, we can't. Without this change, deadlock will occur in testing this PR. However If you can merge split PRs in order, I should be able to do that.
Would you like to split this?

Reply:
I think it should be merged as one PR as the `HTTPServer.swift` change is required for the test case to work.

Reply:
I don't think there is anything specific to `HTTPServer` about the change to `NSData`. How is it using `NSData` that triggers this bug?

Reply:
If the test case that was added (`test_contentsOfURL`) is calling `NSData(contentsOf: url)` and that is running on the main thread then won't the HTTP server have to run on another thread to service the request, or am I mis-understanding how `DispatchQueue` works?

Reply:
This is a code that reproduces this bug.

```swift
import Foundation

if let data = NSData(contentsOf: URL(string: "https://swift.org")!) {
    print(data.length) // 0
}
```

Reply:
Ok, thanks for the additional info.